### ISSUE TITLE
`video` prop unexpectedly appends `dfn` text to link content

### ISSUE THREAD

**Issue by oddball-lindsay** [department-of-veterans-affairs/vets-design-system-documentation#5060]
### Point of contact

Lindsay Li-Smith, Principle Product Manager for the Accredited Representative Crew (ARC)

### Defect Severity

4, Minor. Consider fixing or exploring in 2-4 sprints post-launch.

### Severity rationale

This is minor in that it is not affecting the usability. 

But still something to consider fixing, as the current experience breaks our intended content standards and may affect future external links that we implement. 

Hoping this can be resolved at a Platform level, to avoid the team deviating from DSVA standards and pursuing custom link implementation in the future.

### URL where issue occurs

https://www.va.gov/representative

### Component or pattern (if applicable)

When we implement the `video` prop for a YouTube link, it automatically appends `dfn` text "on Youtube" to the end of the link, which is not ideal or expected for our content. I've attached screenshots of our expected content, and the actual implementation on https://www.va.gov/representative.

### Environment details

This bug is visible across all devices/browsers.

### Steps to reproduce

1. Navigate to https://www.va.gov/representative
2. In the second section "What you can do in the portal", observe the affected link content
3. Inspect the link and you can view the `dfn` content

### Expected behavior / Acceptance criteria

Expected:
<img width="944" height="221" alt="Image" src="https://github.com/user-attachments/assets/656e73bc-f062-4194-8dd9-a17756f1b1f4" />

Actual:
<img width="438" height="151" alt="Image" src="https://github.com/user-attachments/assets/8b75822d-f9d8-4b3b-88a6-9384832406ed" />

### Supporting evidence (screenshots, recordings, code snippets)

Screenshot of the code, where  you can see that `<dfn>on YouTube</dfn>` addition in the code.

<img width="654" height="310" alt="Image" src="https://github.com/user-attachments/assets/aaafa91e-926f-42df-97c2-9b61b384e0d7" />

### Related issues or systemic notes

Initially surfaced to Platform Design in this [Slack conversation](https://dsva.slack.com/archives/C01DBGX4P45/p1761241743084549).

The `dfn` guidance may have originated from this accessibility feedback from 2019:
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/152#issuecomment-553127340

Past discussions indicate the `dfn` text may have been intended for accessibility, but [this post](https://dsva.slack.com/archives/C0335Q88X0S/p1715777739158199) indicates screen readers are not picking up the dfn text so it may not even be functioning as intended. Not everyone may have access to that post, so here's the content:
> The Design system link component has a variants (channel and video) that append text using a <dfn> tag. VO is not reading this text, so I'm not getting necessary information I need to understand the purpose of the link. For example, I have a link going to a Youtube channel and visually it says "Veterans Health Administration YouTube" but on VO all I hear is "Veterans Health Administration". It does work on NVDA. However, is <dfn> really needed here? It seems like maybe aria-label or even just appending it in a <span> might work better?

Some ideas from the ARC team, if we were to deviate from VADS and implement a custom fix:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/121424

[Slack conversation](https://dsva.slack.com/archives/C01DBGX4P45/p1761241743084549)

### Type of issue

- [ ] Axe-core
- [x] Screenreader | assistive tech & device support
- [ ] Keyboard
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Focus
- [ ] Headings
- [x] Content style guide
- [x] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [x] Overall user experience
- [ ] Cognitive | user flows & navigating
- [x] URL standards
- [ ] Something else

**Comment by jeana-adhoc** [comment]
@oddball-lindsay -

When you have a minute, can you say more about why the "on youtube" text
> is not ideal or expected for our content.

It also appears in your screenshot that you're not putting the icon in the link itself? Are you not wanting the icon to be linked either?  

The reason we added "on youtube" is to give the link/icon context and to help the user know where they'd be going - I see it's rather redundant now with the "opens in a new tab" text as well.

---

**Team note:** DST should look at all uses of `<dfn>` to see if it makes sense to use here. `<span>` may be better.

Related links:
- [Storybook](https://design.va.gov/storybook/?path=/story/components-va-link--video)
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2817
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/152#issuecomment-553127340

**Comment by oddball-lindsay** [comment]
@jeana-adhoc `...(opens in new tab) on YouTube` isn't ideal/expected because we already had the `(opens in new tab)` content to set user expectations about the link behavior (albeit less specifically). 

Outside of redundancy, it also feels weird to have the `on YouTube` after the `(opens in new tab)`. If the website mention is important to acknowledge, placement preference is to place it before the parenthesis -- `...on YouTube (opens in new tab)`. 

Mock up: 
<img width="270" height="104" alt="Image" src="https://github.com/user-attachments/assets/c0e225dc-ca82-4d8f-9023-65034900bb93" />

Current/Actual:
<img width="267" height="113" alt="Image" src="https://github.com/user-attachments/assets/8ce2716e-f583-4614-b6d7-fea4d55e1434" />

---

>It also appears in your screenshot that you're not putting the icon in the link itself? Are you not wanting the icon to be linked either?

Here's a screenshot with the code exposed, I think this indicates that the icon is part of the link.

<img width="1249" height="493" alt="Image" src="https://github.com/user-attachments/assets/8119c053-4df9-49df-acaf-3174d863da05" />
